From 297c7d20f0f8d273c8ce2173c502a4aa4a0a374a Mon Sep 17 00:00:00 2001
From: Alfredo Cardigliano <cardigliano@ntop.org>
Date: Wed, 28 Jan 2026 09:57:33 +0100
Subject: [PATCH 6/9] Fix race condition with bond interfaces

---
 kernel/pf_ring.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/pf_ring.c b/kernel/pf_ring.c
index 1e5ec888..87392ec3 100644
--- a/kernel/pf_ring.c
+++ b/kernel/pf_ring.c
@@ -3107,6 +3107,7 @@ static inline int copy_data_to_ring(struct sk_buff *skb,
 #else
     (skb->dev->priv_flags & IFF_EBRIDGE) ||
 #endif
+    (netif_is_bond_master(skb->dev)) ||
     (enable_tx_capture && pfr->direction != rx_only_direction) ||
     (pfr->num_channels_per_ring > 1) ||
     (pfr->channel_id_mask == RING_ANY_CHANNEL && lock_rss_queues(skb->dev)) ||
-- 
2.54.0

