From f91fa816dfabf61684aa6a873bd82c6334ff6ac8 Mon Sep 17 00:00:00 2001
From: Alfredo Cardigliano <cardigliano@ntop.org>
Date: Tue, 24 Feb 2026 10:14:51 +0100
Subject: [PATCH 8/9] Compilation fix on kernel 6.17

---
 kernel/pf_ring.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/kernel/pf_ring.c b/kernel/pf_ring.c
index 87392ec3..7071e7c6 100644
--- a/kernel/pf_ring.c
+++ b/kernel/pf_ring.c
@@ -5469,7 +5469,11 @@ static int is_netdev_promisc(struct net_device *netdev) {
   debug_printk(1, "checking promisc for %s\n", netdev->name);
 
   rtnl_lock();
+#if(LINUX_VERSION_CODE < KERNEL_VERSION(6,17,0))
   if_flags = (short) dev_get_flags(netdev);
+#else
+  if_flags = (short) netif_get_flags(netdev);
+#endif
   rtnl_unlock();
 
   return !!(if_flags & IFF_PROMISC);
@@ -5484,7 +5488,12 @@ static void set_netdev_promisc(struct net_device *netdev) {
 
   rtnl_lock();
 
+#if(LINUX_VERSION_CODE < KERNEL_VERSION(6,17,0))
   if_flags = (short) dev_get_flags(netdev);
+#else
+  if_flags = (short) netif_get_flags(netdev);
+#endif
+
   if(!(if_flags & IFF_PROMISC)) {
     if_flags |= IFF_PROMISC;
 #if(LINUX_VERSION_CODE < KERNEL_VERSION(5,0,0) && \
@@ -5507,7 +5516,12 @@ static void unset_netdev_promisc(struct net_device *netdev) {
 
   rtnl_lock();
 
+#if(LINUX_VERSION_CODE < KERNEL_VERSION(6,17,0))
   if_flags = (short) dev_get_flags(netdev);
+#else
+  if_flags = (short) netif_get_flags(netdev);
+#endif
+
   if(if_flags & IFF_PROMISC) {
     if_flags &= ~IFF_PROMISC;
 #if(LINUX_VERSION_CODE < KERNEL_VERSION(5,0,0) && \
-- 
2.54.0

