--- A/Makefile.in	2017-04-24 11:53:37.000000000 +0200
+++ B/Makefile.in	2017-09-28 09:31:47.426375823 +0200
@@ -110,7 +110,7 @@
 
 ${DYNAMICLIB}: ${OBJS} extract_nbpf @PF_RING_ZC_DEP@ @NT_DEP@ @MYRICOM_DEP@ @DAG_DEP@ @FIBERBLAZE_DEP@ @ACCOLADE_DEP@ @NPCAP_DEP@ pfring.h ${RING_H} Makefile
 	@echo "=*= making library $@ =*="
-	${CC} ${LDFLAGS} ${OBJS} ${NBPF_OBJS} ${SYSLIBS} -o $@
+	${CC} -Wl,-soname,$@.1 ${LDFLAGS} ${OBJS} ${NBPF_OBJS} ${SYSLIBS} -o $@
 
 extract_pfring_zc_lib:
 	@AR_X@ @PF_RING_ZC_LIB@
