--- A/lib/distro_entry.py	2015-09-28 20:39:54.707647791 +0900
+++ B/lib/distro_entry.py	2015-09-28 20:42:45.598511361 +0900
@@ -18,6 +18,9 @@
     "redhat": {"flag_defaults": {"config": "/etc/grr/grr-server.yaml"},
                "config_opts": {"Config.writeback":
                                "/etc/grr/server.local.yaml"}},
+    "gentoo": {"flag_defaults": {"config": "/etc/grr/grr-server.yaml"},
+               "config_opts": {"Config.writeback":
+                               "/etc/grr/server.local.yaml"}},
 }
 
 
@@ -29,6 +32,8 @@
       return "debian"
     if distribution in ["red hat enterprise linux server"]:
       return "redhat"
+    else :
+      return "gentoo"
   raise RuntimeError("Missing distro specific config. Please update "
                      "distro_entry.py.")
 
